packages/@aws-cdk/aws-codepipeline-api/lib/action.ts

```diff
@@ -71,6 +71,25 @@ export interface IStage {
    * @param action the Action to add to this Stage
    */
   _attachAction(action: Action): void;
+
+  /**
+   * Generates a unique output artifact name for the given Action.
+   * This is an internal operation -
+   * you should never need to call it directly.
+   *
+   * @param action the Action to generate the output artifact name for
+   */
+  _generateOutputArtifactName(action: Action): string;
```

Comment:
I wonder if it would be more elegant to hide all these internal methods under some `_internal` scope:

```ts
interface IStageInternal {
  attachAction(...);
  generateOutputArtifactName(...);
  findInputArtifactFor(...);
};

interface IStage {
  // public API

  // internal API
  _internal: IStageInternal;
}
```

Reply:
Done.